### DISCUSSION TITLE
Set Buildbot Web Server to connect to ReadOnly Replica DB

### DISCUSSION THREAD

**Discussion by hlopezvg** [buildbot/buildbot#6773] [General]
Hello, 

I apologize if this is not the right place for this question, but is there a way to connect **buildbot** webserver to a replica DB?

Currently, we have a master **cfg** connected to a read/write DB, but we want to know if WWW can just be set to a replica one. This is the error we got when we tried the above.

`sqlalchemy.exc.InternalError: (psycopg2.errors.ReadOnlySqlTransaction) cannot execute INSERT in a read-only transaction`

Thank you!

**Comment by eighthave** [comment]
This is exactly what we want for F-Droid as well.  Then we can have a secure, headless server that actually runs the builds, then have the website on a separate, isolated server for the public to see everything.  The new [`sqlite_rsync`](https://sqlite.org/rsync.html) tool from SQLite v3.47.0 looks like the perfect tool for the job.

We're tracking it here:
https://gitlab.com/fdroid/wiki/-/wikis/Internal/Buildbot#read-only-website-copy

**Reply by p12tic** [reply]
I think this is worthy a feature request.

**Comment by p12tic** [comment]
Moved to https://github.com/buildbot/buildbot/issues/8252.
